CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)

sdumitriu opened the pull request:
- start in prems mode
- create new visits for the new IC, EDIC, IPIC and EDIPIC clinics
- generate tokens for them and fill them in as a patient
- create a new visit for ED, fill it in and submit as a patient, then edit the VI form and change the clinic to EDIC, generate a new token, check that the patient can see and submit the IC form
- create a new visit for EDIP, save, check that ED+IP forms were created for the visit, then edit the VI form and change the clinic to EDIPIC, generate a token, fill in and submit all 3 forms as a patient (are the 3 forms in the right order?

marta- committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)

sdumitriu committed:
CARDS-2306: Add the ability to override the common header/footer in patient notification emails

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Changed footer in the email template
Added more templates for the 4 different scenarios when the patients had to, but didn't fill in the initial surveys

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Add clinics for ED+IC and ED+IP+IC
Remove conflict mode for the IC clinics to allow adding the IC form to the visit

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Fix IC questionnaire

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Remove "this question is mandatory" instructions from the IC form

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Make the list of questionnaires orderable

sashaandjic commented:
Question: On newly made IC survey, the last question `What else would you like to say about this experience?` is required. On all other surveys this is not the case.
![image](https://github.com/data-team-uhn/cards/assets/23243244/0d122ba4-0f31-4759-b3dc-b672a12062f7)

sashaandjic reviewed (approved):
tested

- [x] create new visits for the new IC, EDIC, IPIC and EDIPIC clinics
- [x] generate tokens for them and fill them in as a patient

- [x] create a new visit for ED, fill it in and submit as a patient, then edit the VI form and change the clinic to EDIC, generate a new token, check that the patient can see and submit the IC form

- [x] create a new visit for EDIP, save, check that ED+IP forms were created for the visit, then edit the VI form and change the clinic to EDIPIC, generate a token, fill in and submit all 3 forms as a patient 

- [x] are the 3 forms in the right order?

marta- commented:
> Question: On newly made IC survey, the last question `What else would you like to say about this experience?` is required. On all other surveys this is not the case.

Typo in the prems import csv, min answers is 1 when it should be 0. Fixed in the csv, will fix in the survey.

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -0,0 +1,730 @@
+		</property>
+		<property>
+			<name>minAnswers</name>
+			<value>1</value>
```
```suggestion
			<value>0</value>
```

The last question should be optional

sashaandjic commented:
> Typo in the prems import csv, min answers is 1 when it should be 0. Fixed in the csv, will fix in the survey.

Argh, that was my bug.

marta- commented:
> > Typo in the prems import csv, min answers is 1 when it should be 0. Fixed in the csv, will fix in the survey.
> 
> Argh, that was my bug.

Nope, mine when I transferred from your spreadsheet to the main one.

sashaandjic reviewed (approved):
tested - the last question is no longer required on IC survey

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Add email sending jobs

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
The "comments" question should not be mandatory

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+    <p>Dear ${first_name} ${last_name},</p>
+    <p>
+      You are receiving this email because of your recent visit at University Health Network (UHN) hospital within the last 30 days.
+      We would like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us
```
```suggestion
      We’d like to hear about your experience while you were a patient in UHN's Emergency Department.
      We would also like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+      team and answering will not impact future care you receive at UHN.
+    </p>
+    <p>
+      The questionnaire contains 7 questions and will take about 5 minutes to complete.
```
```suggestion
      The survey consists of two questionnaires containing 18 questions and will take 10-15 minutes to complete.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+
+    <p>
+      Best regards,<br/>
+      <span class="signature">The Patient Experience Team</span><br/>
```
```suggestion
      <span class="signature">The Integrated Care Team</span><br/>
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+
+    <p>
+      Best regards,<br/>
+      <span class="signature">The Patient Experience Team</span><br/>
```
```suggestion
      <span class="signature">The Integrated Care Team</span><br/>
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC/mailTemplates/InitialInvitation/bodyTemplate.txt:
```diff
@@ -0,0 +1,38 @@
+https://www.uhn.ca/corporate/AboutUHN/Quality_Patient_Safety/Pages/patient_values.aspx.
+
+Best regards,
+The Patient Experience Team
```
```suggestion
The Integrated Care Team
```

marta- reviewed (commented):
I'm suggesting some updates to the email texts, for demo purposes. These have not yet been validated by the client, so let's not change everything until they decide.

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+      </p>
+    </div>
+    <p>
+      If you have any questions, concerns, or wish to be removed from the survey, please contact us at yourexperience@uhn.ca.
```
For all emails inviting patients to fill out an IC survey (even if bundled with other surveys), the contact email address and the reply-to address should be `Ic_Patientexperience@uhn.ca`.

```suggestion
      If you have any questions, concerns, or wish to be removed from the survey, please contact us at Ic_Patientexperience@uhn.ca.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-EDIP/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+    <p>Dear ${first_name} ${last_name},</p>
+    <p>
+      You are receiving this email because of your recent visit at University Health Network (UHN) hospital within the last 30 days.
+      We would like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us
```
```suggestion
      We’d like to hear about your experience while you were a patient at UHN.
      We would also like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-EDIP/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+      team and answering will not impact future care you receive at UHN.
+    </p>
+    <p>
+      The questionnaire contains 7 questions and will take about 5 minutes to complete.
```
```suggestion
      The survey consists of three questionnaires containing 31 questions and will take 15-20 minutes to complete.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-IP/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+      team and answering will not impact future care you receive at UHN.
+    </p>
+    <p>
+      The questionnaire contains 7 questions and will take about 5 minutes to complete.
```
```suggestion
      The survey consists of two questionnaires containing 20 questions and will take about 10-15 minutes to complete.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-IP/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -0,0 +1,49 @@
+    <p>Dear ${first_name} ${last_name},</p>
+    <p>
+      You are receiving this email because of your recent visit at University Health Network (UHN) hospital within the last 30 days.
+      We would like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us
```
```suggestion
      We’d like to hear about your experience while you were an inpatient at UHN.
      We would  also like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/EDIC.xml:
```diff
@@ -0,0 +1,139 @@
+As you look to fill out this survey, you were likely discharged from your stay at a UHN hospital site
+within the last 30 days. We would like you to reflect on your experience with the Integrated Care Program
+as you transitioned home and to provide us with some feedback on supports that you may have received.
+We appreciate any insights you can provide and will look to make improvements based on your feedback.
```
```suggestion
 We’d like to hear about your experience while you were a patient in UHN's Emergency Department.
We would also like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us with some feedback on supports that you may have received.
We appreciate any insights you can provide and will look to make improvements based on your feedback.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/EDIPIC.xml:
```diff
@@ -0,0 +1,187 @@
+As you look to fill out this survey, you were likely discharged from your stay at a UHN hospital site
+within the last 30 days. We would like you to reflect on your experience with the Integrated Care Program
+as you transitioned home and to provide us with some feedback on supports that you may have received.
+We appreciate any insights you can provide and will look to make improvements based on your feedback.
```
```suggestion
We’d like to hear about your experience while you were a patient at UHN.
We would also like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us with some feedback on supports that you may have received.
We appreciate any insights you can provide and will look to make improvements based on your feedback.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/IC.xml:
```diff
@@ -0,0 +1,91 @@
+As you look to fill out this survey, you were likely discharged from your stay at a UHN hospital site
+within the last 30 days. We would like you to reflect on your experience with the Integrated Care Program
+as you transitioned home and to provide us with some feedback on supports that you may have received.
+We appreciate any insights you can provide and will look to make improvements based on your feedback.
```
```suggestion
We would like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us with some feedback on supports that you may have received.
We appreciate any insights you can provide and will look to make improvements based on your feedback.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/IPIC.xml:
```diff
@@ -0,0 +1,139 @@
+As you look to fill out this survey, you were likely discharged from your stay at a UHN hospital site
+within the last 30 days. We would like you to reflect on your experience with the Integrated Care Program
+as you transitioned home and to provide us with some feedback on supports that you may have received.
+We appreciate any insights you can provide and will look to make improvements based on your feedback.
```
```suggestion
We’d like to hear about your experience while you were an inpatient in UHN.
We would also like you to reflect on your experience with the Integrated Care Program as you transitioned home and to provide us with some feedback on supports that you may have received.
We appreciate any insights you can provide and will look to make improvements based on your feedback.
```

marta- reviewed (changes requested):
@sdumitriu I made suggestion comments with the following changes per client's request:
* updated the intro message on the start screen for all IC-related clinics
* updated the intro message and the survey description in the HTML version of the Invitation email for all IC-related clinics - please propagate the changes to the text version as well as the Reminder emails
* for all IC related emails, the contact email address should be `IC_Patientexperience@uhn.ca`, and the signature should be "The Integrated Care Team" (only the HTML version of the EDIC email has these suggestions, please propagate to all others)

Additionally, for all IC related emails, the `reply-to` should be `IC_Patientexperience@uhn.ca`

sashaandjic commented:
Is PR ready for re-test?

marta- commented:
> Is PR ready for re-test?

No, changes are pending.

marta- commented on modules/email-notifications/src/main/java/io/uhndata/cards/emailnotifications/EmailTemplate.java:
I'm approving these changes for CARDS-2306. They could be pulled into a separate PR and merged.

sashaandjic commented:
re-tested
----------

- [ ]  create new visits for the new IC, EDIC, IPIC and EDIPIC clinics

- [ ]  generate tokens for them and fill them in as a patient

- [ ]  create a new visit for ED, fill it in and submit as a patient, then edit the VI form and change the clinic to EDIC, generate a new token, check that the patient can see and submit the IC form

- [ ] create a new visit for EDIP, save, check that ED+IP forms were created for the visit, then edit the VI form and change the clinic to EDIPIC, generate a token, fill in and submit all 3 forms as a patient

- [ ]  are the 3 forms in the right order?

sashaandjic commented:
Incomplete: All 4 `UHN INTEGRATED CARE PATIENT EXPERIENCE SURVEY` grids are missing link in the `Discharge date` column
![image](https://github.com/data-team-uhn/cards/assets/23243244/12a41b08-f4ef-43d5-aa6a-946d70f60b5e)

sashaandjic commented:
**QUESTION**: Why don't we show `ED` surveys from `ED-IC` also in the `ED` grid?
I think it may be useful to see all of the `ED` surveys in one place.

marta- commented:
> **QUESTION**: Why don't we show `ED` surveys from `ED-IC` also in the `ED` grid? I think it may be useful to see all of the `ED` surveys in one place.

I'm not sure I understand this question. Can you elaborate on which grid is not showing the ED surveys from ED+IC?

sashaandjic commented:
Overall Experience column is empty in all grids.

sashaandjic commented:
> I'm not sure I understand this question. Can you elaborate on which grid is not showing the ED surveys from ED+IC?

Here are the steps:
- create an ED visit, fill it in as a patient and submit it
- the ED visit will be shown under UHNED as expected
![image](https://github.com/data-team-uhn/cards/assets/23243244/204b4c9a-5a48-4dff-83de-90445e17fa7a)

- now ad ED-IC to the visit
- UHNED is now empty, and ED visit is now only present in UHNED-IC
![image](https://github.com/data-team-uhn/cards/assets/23243244/58632fbd-4d70-408f-b16d-9719ee73f127)

![image](https://github.com/data-team-uhn/cards/assets/23243244/44f2a081-8278-4761-a022-6e3e62606160)

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Add copy answers configuration to have the columns populated on the dashboard

sdumitriu committed:
CARDS-2311: Add new Integrated Care clinics setup (including surveys and email templates)
Text adjustments based on the client's feedback
Update all email templates to match the same format

marta- reviewed (approved):
The changes look good, did not test yet.

marta- commented:
@sashaandjic Changes to test:
* all emails for surveys including the IC questionnaire
* all welcome messages displayed to the patient on the start screen of surveys including the IC questionnaire
* clinician dasboard - check the IC survey box is now correctly populated
